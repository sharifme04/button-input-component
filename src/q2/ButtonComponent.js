import React from "react";
import PropTypes from "prop-types";

const ButtonComponent = ({
  style = {},
  title = "" || "Button Name",
  isDisabled = false,
  isDanger = false,
  isBlock = false,
  onClick,
  ...props
}) => {
  return (
    <button
      disabled={isDisabled}
      type="button"
      className={`${!isDanger && !isDisabled && "btn btn-primary"}
    ${isDisabled && "btn btn-light"}
    ${isDanger && "btn btn-danger"}
    ${isBlock ? "blockStyle" : "inlineStyle"}
    `}
      onClick={onClick}
    >
      {title}
    </button>
  );
};

ButtonComponent.propTypes = {
  title: PropTypes.string,
  isDisabled: PropTypes.bool,
  isDanger: PropTypes.bool,
  isBlock: PropTypes.bool,
  onClick: PropTypes.func
};

export default ButtonComponent;
