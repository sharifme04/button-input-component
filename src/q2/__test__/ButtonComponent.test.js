import React from "react";
import { shallow } from "enzyme";
import ButtonComponent from "../ButtonComponent";

describe("<InputComponent/>", () => {
  test("return true", () => {
    const component = shallow(<ButtonComponent />);
    expect(component).toMatchSnapshot();
  });

  test("does not break with an empty array", () => {
    const component = shallow(<ButtonComponent />);
    expect(component.find("button")).toHaveLength(1);
  });
});
