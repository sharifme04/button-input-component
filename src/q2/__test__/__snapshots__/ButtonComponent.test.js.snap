// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputComponent/> return true 1`] = `
<button
  className="btn btn-primary
    false
    false
    inlineStyle
    "
  disabled={false}
  type="button"
>
  Button Name
</button>
`;
